engine/graphics/src/vulkan/graphics_vulkan.cpp

```diff
@@ -489,106 +506,114 @@ namespace dmGraphics
                 num_attachments = 2;
             }
 
-            res = CreateFramebuffer(context->m_LogicalDevice.m_Device, context->m_MainRenderPass,
+            res = CreateFramebuffer(vulkan_context->m_LogicalDevice.m_Device, vulkan_context->m_MainRenderPass,
                 swapChain->m_ImageExtent.width, swapChain->m_ImageExtent.height,
-                vk_image_attachments, num_attachments, &context->m_MainFrameBuffers[i]);
+                vk_image_attachments, num_attachments, &vulkan_context->m_MainFrameBuffers[i]);
             CHECK_VK_ERROR(res);
         }
 
         return res;
     }
 
-    VkResult DestroyMainFrameBuffers(HContext context)
+    VkResult DestroyMainFrameBuffers(VulkanContext* vulkan_context)
     {
-        for (uint8_t i=0; i < context->m_MainFrameBuffers.Size(); i++)
+        for (uint8_t i=0; i < vulkan_context->m_MainFrameBuffers.Size(); i++)
         {
-            DestroyFrameBuffer(context->m_LogicalDevice.m_Device, context->m_MainFrameBuffers[i]);
+            DestroyFrameBuffer(vulkan_context->m_LogicalDevice.m_Device, vulkan_context->m_MainFrameBuffers[i]);
         }
-        context->m_MainFrameBuffers.SetCapacity(0);
-        context->m_MainFrameBuffers.SetSize(0);
+        vulkan_context->m_MainFrameBuffers.SetCapacity(0);
+        vulkan_context->m_MainFrameBuffers.SetSize(0);
         return VK_SUCCESS;
     }
 
-    static VkResult SetupMainRenderTarget(HContext context)
+    static VkResult SetupMainRenderTarget(VulkanContext* vulkan_context)
     {
         // Initialize the dummy rendertarget for the main framebuffer
         // The m_Framebuffer construct will be rotated sequentially
         // with the framebuffer objects created per swap chain.
-        RenderTarget& rt          = context->m_MainRenderTarget;
-        rt.m_RenderPass           = context->m_MainRenderPass;
-        rt.m_Framebuffer          = context->m_MainFrameBuffers[0];
-        rt.m_Extent               = context->m_SwapChain->m_ImageExtent;
-        rt.m_ColorAttachmentCount = 1;
+        RenderTarget* rt           = new RenderTarget(DM_RENDERTARGET_BACKBUFFER_ID);
+        rt->m_RenderPass           = vulkan_context->m_MainRenderPass;
+        rt->m_Framebuffer          = vulkan_context->m_MainFrameBuffers[0];
+        rt->m_Extent               = vulkan_context->m_SwapChain->m_ImageExtent;
+        rt->m_ColorAttachmentCount = 1;
+
+        VulkanSharedAsset* asset = new VulkanSharedAsset();
+        asset->m_Type            = RESOURCE_TYPE_RENDER_TARGET;
+        asset->m_Asset           = rt;
```

Comment:
It's a bit annoying to do more allocations now with the shared asset indirection that ends up in the container, maybe it's worth to pack the type into the handle 🤔 Eventually when (if) we change the render script API to generalize the texture functions to bind a texture or render target by handle we need to have the type information I think